src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheCrusader/boss_northrend_beasts.cpp

```diff
@@ -295,6 +306,58 @@ class boss_gormok : public CreatureScript
         }
 };
 
+class SnobolledTargetSelector : public std::unary_function<Unit*, bool>
+{
+public:
+    SnobolledTargetSelector(Unit const* unit, float dist = 200.0f) : _me(unit), _dist(dist) { }
+
+    bool operator()(Unit* unit) const
+    {
+        if (unit->GetTypeId() != TYPEID_PLAYER)
+            return false;
+
+        if (_dist > 0.0f && !_me->IsWithinCombatRange(unit, _dist))
+            return false;
+
+        if (!_me->IsWithinCombatRange(unit, -_dist))
+            return false;
+
+        if (unit->HasAura(SPELL_RIDE_PLAYER) || unit->HasAura(SPELL_SNOBOLLED))
+            return false;
+
+        return true;
+    }
+
+private:
+    Unit const* _me;
+    float _dist;
+};
+
+class SnoboldEvent : public BasicEvent
+{
+public:
+    SnoboldEvent(Unit* owner, Unit* target, bool apply) : _owner(owner), _target(target), _apply(apply) { }
+
+    bool Execute(uint64 /*eventTime*/, uint32 /*updateTime*/) override
+    {
+        if (_apply)
+            _owner->CastSpell(_target, SPELL_SNOBOLLED, true);
+
+        else
+        {
+            _target->RemoveAurasDueToSpell(SPELL_SNOBOLLED);
+            _target->RemoveVehicleKit();
+        }
+
+        return true;
+    }
+
+private:
+    Unit* _owner;
```

Comment:
Removing Assync Call, gets me alot of issues with spell cast fails.
I'm searching motive with sirkfoll, but no make any sense.
btw, we are working on find why some spellcast fail when remove assync call